File: typescript/smily.ts
/**
 * 
 * @param texts list of string [":)", ":(", ":>"]
 * @returns amount of smily face detected
 */
export function countSmilyFace(texts: string[]) : number {
    // TODO : Start your code here
    const validSmileyRegex = /^[:;][-~]?[)D]$/;
    return texts.filter(face => validSmileyRegex.test(face)).length;
    // throw new Error("Not implemented");
}

File: golang/smily.go
package main

import "regexp"

func CountSmilyFace(text []string) int {
	// TODO : start your code here
	count := 0
	validSmileyRegex := regexp.MustCompile(`^[:;][-~]?[)D]$`)

	for _, face := range text {
		if validSmileyRegex.MatchString(face) {
			count++
		}
	}

	return count
}
